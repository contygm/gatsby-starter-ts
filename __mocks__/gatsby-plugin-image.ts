import React from 'react';
const plugin = jest.requireActual('gatsby-plugin-image');

const mockImage = ({ imgClassName, ...props } : {imgClassName: string }) =>
    React.createElement('img', {
        ...props,
        className: imgClassName
    });

module.exports = {
    ...plugin,
    GatsbyImage: jest.fn().mockImplementation(mockImage),
    StaticImage: jest.fn().mockImplementation(mockImage)
};
