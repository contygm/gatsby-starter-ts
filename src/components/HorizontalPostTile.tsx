import { Link } from 'gatsby';
import { GatsbyImage, IGatsbyImageData } from 'gatsby-plugin-image';
import * as React from 'react';

export interface PostTileProps {
    title: string;
    excerpt: string;
    slug: string;
    image?: IGatsbyImageData;
}

const HorizontalPostTile = ({ title, excerpt, slug, image }: PostTileProps) => {
    return (
        <Link
            to={`/blog${slug}`}
            className="horizontal-post-tile"
        >
            <div className="media">
                {image && (
                    <figure className="media-left my-auto">
                        <GatsbyImage
                            alt="blog image"
                            image={image}
                            className="image is-64x64 post-tile-image"
                        />
                    </figure>
                )}
                <div className="media-content">
                    <div className="content">
                        <h6>{title}</h6>
                        <p>{excerpt}</p>
                    </div>
                </div>
            </div>
        </Link>
    );
};

export default HorizontalPostTile;
