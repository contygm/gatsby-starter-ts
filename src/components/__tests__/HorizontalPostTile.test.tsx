import React from 'react';
import { render } from '@testing-library/react';
import HorizontalPostTile, { PostTileProps } from '../HorizontalPostTile';
import { IGatsbyImageData } from 'gatsby-plugin-image';

const mockData: PostTileProps = {
    title: 'A Blog Title',
    excerpt: 'Cheesecake lemon drops cheesecake lemon drops pastry danishâ€¦',
    slug: '/blog-url/',
    image: {} as IGatsbyImageData
};

describe('HorizontalPostTile', () => {
    it('renders correctly', () => {
        const { asFragment } = render(<HorizontalPostTile {...mockData} />);
        expect(asFragment()).toMatchSnapshot();
    });
});
