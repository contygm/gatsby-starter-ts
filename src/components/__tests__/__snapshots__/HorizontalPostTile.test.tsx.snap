// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HorizontalPostTile renders correctly 1`] = `
<DocumentFragment>
  <a
    class="horizontal-post-tile"
    href="/blog/blog-url/"
  >
    <div
      class="media"
    >
      <figure
        class="media-left my-auto"
      >
        <img
          alt="blog image"
          image="[object Object]"
        />
      </figure>
      <div
        class="media-content"
      >
        <div
          class="content"
        >
          <h6>
            A Blog Title
          </h6>
          <p>
            Cheesecake lemon drops cheesecake lemon drops pastry danishâ€¦
          </p>
        </div>
      </div>
    </div>
  </a>
</DocumentFragment>
`;
